File: go/src/stream_consumer/src/events.go
package streamConsumer

import (
	"github.com/aws/aws-lambda-go/events"
	"github.com/sirupsen/logrus"
)

func SendEvents(record events.DynamoDBEventRecord) {
	switch {
	case isNewDeposit(record):
		sendEvent("deposit-requested", record.Change.NewImage)
	case isDepositSent(record):
		sendEvent("deposit-sent", record.Change.NewImage)
	}
}

func sendEvent(detailType string, item map[string]events.DynamoDBAttributeValue) {
	// TODO: replace with real event publication (EventBridge PutEvents)
	logrus.WithFields(logrus.Fields{
		"detailType": detailType,
		"depositId":  item["depositId"].String(),
	}).Info("event would be published")
}

File: python/src/stream_consumer/events.py
import logging
from typing import Any

log = logging.getLogger(__name__)


def send_event(detail_type: str, item: dict[str, Any]) -> None:
    # mock event publication (EventBridge PutEvents)
    deposit_id = item.get("depositId", {}).get("S", "")
    log.info(
        "event would be published",
        extra={"detailType": detail_type, "depositId": deposit_id},
    )
